lib/zlib.js

```diff
@@ -165,95 +141,112 @@ function flushCallback(level, strategy, callback) {
+        if (finishFlush < Z_NO_FLUSH || finishFlush > Z_BLOCK ||
+            !Number.isFinite(finishFlush)) {
+          throw new RangeError('Invalid flush flag: ' + finishFlush);
+        }
+      } else {
+        finishFlush = Z_FINISH;
       }
-    }
 
-    if (opts.windowBits !== undefined) {
-      if (opts.windowBits < constants.Z_MIN_WINDOWBITS ||
-          opts.windowBits > constants.Z_MAX_WINDOWBITS) {
-        throw new RangeError('Invalid windowBits: ' + opts.windowBits);
+      windowBits = opts.windowBits;
+      if (windowBits !== undefined && windowBits === windowBits) {
+        if (windowBits < Z_MIN_WINDOWBITS || windowBits > Z_MAX_WINDOWBITS ||
+            !Number.isFinite(windowBits)) {
+          throw new RangeError('Invalid windowBits: ' + windowBits);
+        }
+      } else {
+        windowBits = Z_DEFAULT_WINDOWBITS;
       }
-    }
 
-    if (opts.level !== undefined) {
-      if (opts.level < constants.Z_MIN_LEVEL ||
-          opts.level > constants.Z_MAX_LEVEL) {
-        throw new RangeError('Invalid compression level: ' + opts.level);
+      level = opts.level;
+      if (level !== undefined && level === level) {
+        if (level < Z_MIN_LEVEL || level > Z_MAX_LEVEL ||
+            !Number.isFinite(level)) {
+          throw new RangeError('Invalid compression level: ' + level);
+        }
+      } else {
+        level = Z_DEFAULT_COMPRESSION;
       }
-    }
 
-    if (opts.memLevel !== undefined) {
-      if (opts.memLevel < constants.Z_MIN_MEMLEVEL ||
-          opts.memLevel > constants.Z_MAX_MEMLEVEL) {
-        throw new RangeError('Invalid memLevel: ' + opts.memLevel);
+      memLevel = opts.memLevel;
+      if (memLevel !== undefined && memLevel === memLevel) {
+        if (memLevel < Z_MIN_MEMLEVEL || memLevel > Z_MAX_MEMLEVEL ||
+            !Number.isFinite(memLevel)) {
+          throw new RangeError('Invalid memLevel: ' + memLevel);
+        }
+      } else {
+        memLevel = Z_DEFAULT_MEMLEVEL;
       }
-    }
 
-    if (opts.strategy !== undefined && isInvalidStrategy(opts.strategy))
-      throw new TypeError('Invalid strategy: ' + opts.strategy);
+      strategy = opts.strategy;
+      if (strategy !== undefined && strategy === strategy) {
+        if (strategy < Z_DEFAULT_STRATEGY || strategy > Z_FIXED ||
+            !Number.isFinite(strategy)) {
+          throw new TypeError('Invalid strategy: ' + strategy);
+        }
+      } else {
+        strategy = Z_DEFAULT_STRATEGY;
+      }
 
-    if (opts.dictionary !== undefined) {
-      if (!ArrayBuffer.isView(opts.dictionary)) {
+      dictionary = opts.dictionary;
+      if (dictionary !== undefined && !ArrayBuffer.isView(dictionary)) {
         throw new TypeError(
           'Invalid dictionary: it should be a Buffer, TypedArray, or DataView');
       }
-    }
 
+      if (opts.encoding || opts.objectMode || opts.writableObjectMode) {
+        opts = _extend({}, opts);
```

Comment:
I don't have any handy, no. We could always flip it to be a whitelist in the future I suppose if we run into issues.